### PR TITLE
Upgrade minimum Python version to 3.8

### PR DESCRIPTION
**Pull request by saulfield**
This PR bumps the minimum supported Python version to 3.8. This has been planned for some time, but was made necessary during the Argoverse 2 integration which requires >=3.8.

Some package version changes were needed:
- Upgraded `ray[rllib]` from `1.0.1.post1` to `1.4.0` to get rid of `gym[atari]` dependency which had problems installing in 3.8
- Put a `numpy<1.24.0` upper bound to fix an issue with `ray[rllib]==1.4.0` (see https://github.com/ray-project/ray/issues/31258)

Note also that an issue with notebooks popped up: #1902. We have decided to ignore the test for now to unblock the Argoverse integration, but this should be investigated.

### CODE REVIEW

**Comment by Gamenot** [comment]
Please add a changelog entry.

File: utils/setup/install_deps.sh [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
     || hash python3.9 2>/dev/null;
 }
 
 function install_python_3_7 {
-    echo "Installing python3.7"
+    echo "Installing python3.8"
     sudo apt-get install $1 software-properties-common
     sudo add-apt-repository $1 ppa:deadsnakes/ppa
-    sudo apt-get install $1 python3.7 python3.7-tk python3.7-venv
+    sudo apt-get install $1 python3.8 python3.8-tk python3.8-venv
```
Comment (Gamenot) [inline utils/setup/install_deps.sh:15]: Do you know why we need python Tkinter? It is not in most other requirements.

Comment (saulfield) [reply utils/setup/install_deps.sh:15]: I'm not sure. It seems it's been there for a while.

File: utils/setup/install_deps.sh [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
     # running through current minor versions
-    hash python3.7 2>/dev/null \
+    hash python3.8 2>/dev/null \
     || hash python3.8 2>/dev/null \
     || hash python3.9 2>/dev/null;
 }
 
 function install_python_3_7 {
-    echo "Installing python3.7"
+    echo "Installing python3.8"
```
Comment (Adaickalavan) [inline utils/setup/install_deps.sh:12]: Maybe the function names can also be update, such as `function install_python_3_8` .

**Review by Adaickalavan — APPROVED** [review]

**Comment by Gamenot** [comment]
Is there anything blocking this?

**Pull request merged**